### ISSUE TITLE
Warn about outdated `actions/cache` use

### ISSUE THREAD

**Issue by jenstroeger** [rhysd/actionlint#622]
Hi,  I have a few actions that use
```yaml
uses: actions/cache@2b250bc32ad02700b996b496c14ac8c2840a2991 # v2.1.8 
```
which runs clean through actionlint v1.7.11.24 but then raises this error on Github:

![Image](https://github.com/user-attachments/assets/60ca48ed-2107-4e35-a3ac-cb4842e905c5)

I think it would be good if actionlint could flag this.

**Comment by rhysd** [comment]
I think dependabot works better here because it can suggest the fix.

It is not obvious when to warn outdated actions properly. For example, the latest version is sometimes not available in self-hosted runners. Simple warnings can cause false positives in such cases.

**Comment by jenstroeger** [comment]
I mentioned it because actionlint already checks other versions, for example I got this today:
```
.github/workflows/some-action.yaml:25:13: the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue [action]
   |
25 |       uses: actions/checkout@v3
   |             ^~~~~~~~~~~~~~~~~~~
```

**Comment by muzimuzhi** [comment]
@jenstroeger What you saw is the check [Outdated popular actions detection at `uses:`](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#outdated-popular-actions-detection-at-uses):

> [...] An action is outdated when the runner used by the action is no longer supported by GitHub Actions runtime. For example, node12 is no longer available so any actions can not use node12 runner.
>
> Note that this check doesn't report that the action version is up-to-date.

Similar to the check [Popular action inputs validation at `with:`](https://github.com/rhysd/actionlint/blob/main/docs/checks.md#popular-action-inputs-validation-at-with), it seems the detection for outdated popular action also

> only supports the case of specifying major versions like `actions/checkout@v4`.

Therefore `uses: actions/cache@v2` would raise an outdated version warning, but your commit SHA form passed.

```yaml
uses: actions/cache@2b250bc32ad02700b996b496c14ac8c2840a2991 # v2.1.8
```

**Comment by jenstroeger** [comment]
Thanks @muzimuzhi for explaining the details, that makes sense. If this is something you guys _don’t_ plan to implement, feel free to close this issue.
